### ISSUE TITLE
UnusedFormalParameter of closure is incorrectly detected as UnusedLocalVariable

### ISSUE THREAD

**Issue by tominventisbe** [phpmd/phpmd#580]
When having a parameter in a closure that is not being used, PHPMD reports it as `UnusedLocalVariable` rather than `UnusedFormalParameter`, for example:

```php
function ($blah) {

};
```

Here PHPMD will indicate that `$blah` is an unused local variable, but this behavior is inconsistent with other functions and methods, as unused parameters there are only reported when the `UnusedFormalParameter` rule is active.

This happens on latest master (and likely also on 2.6.0 as the rule doesn't appear to have any behavioral changes since then).

**Comment by kylekatarnls** [comment]
As per my unit tests with the last version UnusedFormalParameter is triggered too. So it actually trigger both rule instead of just UnusedFormalParameter.

**Comment by kylekatarnls** [comment]
Hello, I fixed it. You can test the fix requiring `"phpmd/phpmd": "dev-feature/issue-580-clsoure-formal-parameter"` in your **composer.json**.

It might be released in 2.9.0.

**Comment by tvbeek** [comment]
@tominventisbe did you have the possibility to very that the change is good fix? It can always help the maintainers to get a verification from the reporter 😃

**Comment by tominventisbe** [comment]
@tvbeek I tried the branch and can confirm the error message is now the expected one. Thanks!

**Comment by tvbeek** [comment]
@tominventisbe thanks for the confirmation 👍
